sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1821,6 +1823,23 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
     new AnalysisException(s"Table or view not found: $table")
   }
 
+  def noSuchFunctionError(
```

Comment:
We already have a `def noSuchFunctionError` and it's error message looks pretty good. I think we just need to append a hint for using `SELECT DISTINCT ...` if the function name is a single `DISTINCT`.

Reply:
I see why it makes sense to merge the SELECT DISTINCT hint with existing undefined function error message.

The difference comes from how to propose a new error class where we have to think about the scenario: why this happen, what could be the root cause, what are the mitigations. 

This way of thinking differs both scenarios:
1. distinct could be wrong usage of deduplicating arguments or catalog.db.distinct not exist or users use a wrong name for the registered function (the one to use might be `distincy` etc.
2. undefined function is fun(x) not found, very clear.

I see there is some overlaps. But the scenarios have difference if we dive into it.